docs/_posts/examples/3400-01-14-filter-features-within-map-view.html

```diff
@@ -149,31 +155,35 @@
         "type": "symbol",
         "layout": {
             "icon-image": "airport-15",
-            "icon-padding": 0
-        },
-        "filter": ["in", "abbrev", ""]
+            "icon-padding": 0,
+            "icon-allow-overlap":true
+        }
     });
-
     map.on('moveend', function() {
-        var features = map.querySourceFeatures('airports', {
-            sourceLayer: 'ne_10m_airports'
-        });
+        var features = map.queryRenderedFeatures({layers:['airport']});
 
         if (features) {
+            var renderedAbbrevs = {};
+            // It is possible for the same feature to be included twice in the output of
+            // `map.queryRenderedFeatures()` if features span tile boundaries, so we
+            // filter them out by a unique key
+            var uniqueFeatures = features.filter(function(f) {
+                if (renderedAbbrevs[f.properties.abbrev]) {
+                    return false;
+                } else {
+                    renderedAbbrevs[f.properties.abbrev] = true;
+                    return true;
+                }
+            });
```

Comment:
This code would be more readable if you factored this code into a separate utility function

```js
function unique(array, comparator) { ... }
```